docs/data/core-edit-post.md

```diff
@@ -0,0 +1,271 @@
+
+*Parameters*
+
+ * state: Global application state.
+
+*Returns*
+
+Preferences Object.
+
+### getPreference
+
+
+
+*Parameters*
+
+ * state: Global application state.
+ * preferenceKey: Preference Key.
+ * defaultValue: Default Value.
+
+*Returns*
+
+Preference Value.
+
+### isPublishSidebarOpened
+
+Returns true if the publish sidebar is opened.
+
+*Parameters*
+
+ * state: Global application state
+
+*Returns*
+
+Whether the publish sidebar is open.
+
+### isEditorSidebarPanelOpened
+
+Returns true if the editor sidebar panel is open, or false otherwise.
+
+*Parameters*
+
+ * state: Global application state.
+ * panel: Sidebar panel name.
+
+*Returns*
+
+Whether the sidebar panel is open.
+
+### isFeatureActive
+
+Returns whether the given feature is enabled or not.
+
+*Parameters*
+
+ * state: Global application state.
+ * feature: Feature slug.
+
+*Returns*
+
+Is active.
+
+### isPluginItemPinned
+
+Returns true if the the plugin item is pinned to the header.
+When the value is not set it defaults to true.
+
+*Parameters*
+
+ * state: Global application state.
+ * pluginName: Plugin item name.
+
+*Returns*
+
+Whether the plugin item is pinned.
+
+### getMetaBoxes
+
+Returns the state of legacy meta boxes.
+
+*Parameters*
+
+ * state: Global application state.
+
+*Returns*
+
+State of meta boxes.
+
+### getMetaBox
+
+Returns the state of legacy meta boxes.
+
+*Parameters*
+
+ * state: Global application state.
+ * location: Location of the meta box.
+
+*Returns*
+
+State of meta box at specified location.
+
+### isSavingMetaBoxes
+
+Returns true if the the Meta Boxes are being saved.
+
+*Parameters*
+
+ * state: Global application state.
+
+*Returns*
+
+Whether the metaboxes are being saved.
+
+## Actions
+
+### openGeneralSidebar
+
+Returns an action object used in signalling that the user opened an editor sidebar.
+
+*Parameters*
+
+ * name: Sidebar name to be opened.
+
+
+### closeGeneralSidebar
+
+Returns an action object signalling that the user closed the sidebar.
+
+
+
+
+### openPublishSidebar
+
+Returns an action object used in signalling that the user opened the publish
+sidebar.
+
+
+
+
+### closePublishSidebar
+
+Returns an action object used in signalling that the user closed the
+publish sidebar.
+
+
+
+
+### togglePublishSidebar
+
+Returns an action object used in signalling that the user toggles the publish sidebar.
+
+
+
+
+### toggleGeneralSidebarEditorPanel
+
+Returns an action object used in signalling that use toggled a panel in the editor.
+
+*Parameters*
+
+ * panel: The panel to toggle.
+
+
+### toggleFeature
+
+Returns an action object used to toggle a feature flag.
+
+*Parameters*
+
+ * feature: Feature name.
+
+
+### togglePinnedPluginItem
+
+Returns an action object used to toggle a plugin name flag.
+
+*Parameters*
+
+ * pluginName: Plugin name.
+
+
+### initializeMetaBoxState
+
+Returns an action object used to check the state of meta boxes at a location.
```

Comment:
This is an example of where the extra detail and explanation really helps. A lot of the docs around these selectors is stating the obvious or just re-stating their names and function method in a more verbose and manual way.

Maybe it's worth not adding redundant docs for a lot of them if it's very obvious.